test/parallel_api/algorithm/alg.sorting/sort.pass.cpp

```diff
@@ -451,7 +451,7 @@ main()
             [](size_t k, size_t val) {
             return std::int16_t(val) * (k % 2 ? 1 : -1); });
 
-#if TEST_DPCPP_BACKEND_PRESENT
+#if TEST_DPCPP_BACKEND_PRESENT && !ONEDPL_FPGA_DEVICE
```

Comment:
Probably only some FPGA-devices hasn't `sycl::half` support.
If so, required to check device aspect about this.